### DISCUSSION TITLE
Help me install Seurat 5

### DISCUSSION THREAD

**Discussion by wykpenguin** [satijalab/seurat#7150] [Q&A]
I am running R studio Version 2023.03.0+386 with R version 4.3.0 beta (2023-04-08 r84203)

`install.packages('Seurat')` works fine and I seem to download a bunch of packages
```
Installing package into ‘/Users/[username]/Library/R/x86_64/4.3/library’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/macosx/big-sur-x86_64/contrib/4.3/Seurat_4.3.0.tgz'
Content type 'application/x-gzip' length 3774641 bytes (3.6 MB)
==================================================
downloaded 3.6 MB


The downloaded binary packages are in
	/var/folders/v2/_b6klrh57w9c3y3h9dm5g1qh0000gt/T//RtmpXnKCaW/downloaded_packages
```

but when I use `library(seurat)` I get the following error:
`Error: package or namespace load failed for ‘Seurat’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/[username]/Library/R/x86_64/4.3/library/sctransform/libs/sctransform.so':
  dlopen(/Users/[username]/Library/R/x86_64/4.3/library/sctransform/libs/sctransform.so, 0x0006): Library not loaded: '/opt/gfortran/lib/gcc/x86_64-apple-darwin20.0/12.2.0/libgfortran.5.dylib'
  Referenced from: '/Users/[username]/Library/R/x86_64/4.3/library/sctransform/libs/sctransform.so'
  Reason: tried: '/opt/gfortran/lib/gcc/x86_64-apple-darwin20.0/12.2.0/libgfortran.5.dylib' (no such file), '/usr/local/lib/libgfortran.5.dylib' (no such file), '/usr/lib/libgfortran.5.dylib' (no such file)`

When I look at my directory, there is not a Library folder there either. What did I do wrong? Many thanks.

**Comment by wudustan** [comment]
It looks like you need to install Fortran. Since it looks like you're on MacOS you can use Homebrew to install it `brew install gcc
`. 

You'll need to have Xcode installed first. You might also need to modify your R Makevars file to specify the installed gcc location. Google has a bunch of guides on how to do all this but HMU if you need a hand.

**Comment by brandenjlynch** [comment]
I am seeing the same issue. I am running an Intel based Mac, with R 4.2.2, and have installed Seurat 4.3.01. I installed Xcode and gfortran 12.2 from the official R-development page for Mac (https://mac.r-project.org/tools/). I tried creating ~/.R/Makevars:

`F77=/opt/gfortran/bin/gfortran
FC=/opt/gfortran/bin/gfortran
FLIBS=-L/opt//gfortran/lib/gcc/x86_64-apple-darwin20.0/12.2.0 -lgfortran`

I also tried manually loading the library:
`
> dyn.load("/opt/gfortran/lib/gcc/x86_64-apple-darwin20.0/12.2.0/libgfortran.5.dylib")
> library(Seurat)
Error: package or namespace load failed for ‘Seurat’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/user/Library/Caches/org.R-project.R/R/renv/cache/v5/R-4.2/x86_64-apple-darwin17.0/irlba/2.3.5.1/acb06a47b732c6251afd16e19c3201ff/irlba/libs/irlba.so':
  dlopen(/Users/user/Library/Caches/org.R-project.R/R/renv/cache/v5/R-4.2/x86_64-apple-darwin17.0/irlba/2.3.5.1/acb06a47b732c6251afd16e19c3201ff/irlba/libs/irlba.so, 0x0006): Library not loaded: /osxcross/bin/../x86_64-apple-darwin22/lib/libgfortran.5.dylib
  Referenced from: <221020A0-FF6C-3BFA-96F7-BFAE2626DCE4> /Users/user/Library/Caches/org.R-project.R/R/renv/cache/v5/R-4.2/x86_64-apple-darwin17.0/irlba/2.3.5.1/acb06a47b732c6251afd16e19c3201ff/irlba/libs/irlba.so
  Reason: tried: '/osxcross/bin/../x86_64-apple-darwin22/lib/libgfortran.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/osxcross/bin/../x86_64-apple-darwin22/lib/libgfortran.5.dylib' (no such file), '/osxcross/bin/../`

I'd be glad to hear any suggestions you might have! I am doing all of this inside an renv, and my feeling is that renv may handle the Makevars file differently (see the changelog for renv 0.8.3): https://rstudio.github.io/renv/news/index.html
However, I haven't been able to figure out how/where to alter renv's handling of Makevars. Still, one wonders why loading the file directly isn't sufficient.

**Reply by brandenjlynch** [reply]
For anyone seeing the same issue -- I uninstalled Seurat and irlba, then deleted their respective directories in my renv. I then reinstalled each with rebuild = TRUE, and voila... no more issues. I didn't alter my Makevars or anything else.
